src/types.ts

```diff
@@ -926,17 +926,27 @@ export class Signature {
   }
 
   /** Tests if a value of this function type is assignable to a target of the specified function type. */
-  isAssignableTo(target: Signature): bool {
-
-    // check `this` type
+  isAssignableTo(target: Signature, checkCompatibleOverride: bool = false): bool {
     var thisThisType = this.thisType;
     var targetThisType = target.thisType;
-    if (thisThisType) {
-      if (!targetThisType || !thisThisType.isAssignableTo(targetThisType)) {
+    if (!checkCompatibleOverride) {
+      // check exact `this` type
+      if (thisThisType) {
+        if (!targetThisType || !thisThisType.isAssignableTo(targetThisType)) {
+          return false;
+        }
+      } else if (targetThisType) {
+        return false;
+      }
+    } else {
+      // check kind of `this` type
+      if (thisThisType) {
+        if (!targetThisType || thisThisType.kind != targetThisType.kind || thisThisType.isReference != targetThisType.isReference) {
+          return false;
+        }
```

Comment:
Here we still need to make sure that the `this` type matches enough so having an overload will always result in a functioning callable signature. Using the `kind` here, so `this` is both times `usize` for example, and both times either a reference or not. Likely needs more checks, so managed and unmanaged aren't mixed etc., which could then become a separate helper function.

Reply:
Hi Dcode,
how about the below function? for the multi implements / overrides cases, do we only need to consider with reference type?
```
  isLenientlyAssignableTo(target: Type) {
    if (target) {
      if (this.isReference && target.isReference) {
        let thisClass = this.getClass();
        let targetClass = target.getClass();
        if (thisClass && targetClass) {
          // extends ThisClass implements TargetInterface
          // implements ThisInterface, TargetInterface
          if (thisClass.kind == ElementKind.CLASS || thisClass.kind == ElementKind.INTERFACE) {
            if (targetClass.kind == ElementKind.INTERFACE) {
              return true;
            }
          }
        }
      }
    }
    return false;
  }
```
@dcodeIO

Reply:
Needs to be a little more complex I think. Perhaps `.isInternalReference`, so external and internal aren't mixed. We'd also need to check that unmanaged and managed aren't mixed (`type.isManaged == type.isManaged`), since both are classes. And the method in question might have an annotated `this` type on its own, so it makes sense to check the method, not just the class, hmm.

Reply:
Hi Dcode,
yeah, I changed the name from `isLenientlyAssignableTo` to `isCompatibleTo`, so it make sense now, please have a look the below method:
```
  /** Tests if a value of this class is compatible to the target class / interface in the multi extends / implements situation. */
  isCompatibleTo(target: Type) {
    if (target) {
      if (this.isInternalReference && target.isInternalReference && this.isManaged && target.isManaged) {
        let thisClass = this.getClass();
        let targetClass = target.getClass();
        if (thisClass && targetClass) {
          // extends ThisClass implements TargetInterface
          // implements ThisInterface, TargetInterface
          if (thisClass.kind == ElementKind.CLASS || thisClass.kind == ElementKind.INTERFACE) {
            if (targetClass.kind == ElementKind.INTERFACE) {
              return true;
            }
          }
        }
      }
    }
    return false;
  }
```

Reply:
Hi Dcode,
I pushed the changes, can you take a look? thank you.
@dcodeIO